src/core/stream.js

```diff
@@ -455,7 +458,11 @@ function Stream({
   const endOfPlay = timings
     .filter(({ ts, duration }) => (
       duration > 0 &&
-      Math.min(duration, timeFragment.end) - ts < END_OF_PLAY
+      Math.min(
```

Comment:
ugly as fugg but makes sense here.

We're stopping when we reach either:
  - the duration
  - the duration in the manifest
  - the end in the timeFragment given